### ISSUE TITLE
Can't compile using Clang on Windows

### ISSUE THREAD

**Issue by madyanov** [glfw/glfw#1807]
Clang 10.0
Compiling on Windows targeting Windows

After updating from revision 96f9f5c4b970140ab5380fb9886ba787c8698937 to current 3327050ca66ad34426a82c217c2d60ced61526b7

```
error: no such file or directory: '/clang:-Wall'

C:\PROGRA~1\LLVM\bin\clang.exe -D_GLFW_USE_CONFIG_H -D_UNICODE -ID:/Projects/l2mapconv-public/libs/glfw/include -ID:/Projects/l2mapconv-public/libs/glfw/src -Ilibs/glfw/src -g -Xclang -gcodeview -O0 -D_DEBUG -D_DLL -D_MT -Xclang --dependent-lib=msvcrtd   /clang:-Wall -MD -MT libs/glfw/src/CMakeFiles/glfw.dir/context.c.obj -MF libs\glfw\src\CMakeFiles\glfw.dir\context.c.obj.d -o libs/glfw/src/CMakeFiles/glfw.dir/context.c.obj   -c D:/Projects/l2mapconv-public/libs/glfw/src/context.c
```

I think issue caused by this commit: https://github.com/glfw/glfw/commit/ac627706ef93362caf08bd3bf249656efd9151e8

**Comment by madyanov** [comment]
I've change the line in src/CMakeLists.txt

`if ("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")`

to

`if ("${CMAKE_C_COMPILER_FRONTEND_VARIANT}" STREQUAL "MSVC")`

and it compiles now.

But I not sure if it broke clang-cl support.
Also `CMAKE_C_COMPILER_FRONTEND_VARIANT` is supported only in CMake 3.15+ (https://gitlab.kitware.com/cmake/cmake/-/merge_requests/3650/diffs)

**Comment by madyanov** [comment]
I've changed the line in src/CMakeLists.txt

`if ("${CMAKE_C_SIMULATE_ID}" STREQUAL "MSVC")`

to

`if ("${CMAKE_C_COMPILER_FRONTEND_VARIANT}" STREQUAL "MSVC")`

and it compiles now.

But I'm not sure if it broke the clang-cl support.
Also `CMAKE_C_COMPILER_FRONTEND_VARIANT` supported only on CMake 3.15+ I think.

**Comment by alxdb** [comment]
Hey all, I've just had this exact same issue. The aforementioned fix works for me as well (change `CMAKE_C_SIMULATE_ID` to `CMAKE_C_COMPILER_FRONTEND_VARIANT`)

**Comment by elmindreda** [comment]
This should be fixed now with 061a0263a9783c1442ad96a061c717c167ab4a76. Thank you for reporting this!

**Comment by alxdb** [comment]
Thank you for fixing it! 🙌
